pkg/controllers/managementuser/clusterauthtoken/cluster_auth_token.go

```diff
@@ -0,0 +1,90 @@
+package clusterauthtoken
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/rancher/rancher/pkg/auth/tokens"
+	mgmtcontrollers "github.com/rancher/rancher/pkg/generated/controllers/management.cattle.io/v3"
+	clusterv3 "github.com/rancher/rancher/pkg/generated/norman/cluster.cattle.io/v3"
+	"github.com/sirupsen/logrus"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type clusterAuthTokenHandler struct {
+	tokenCache  mgmtcontrollers.TokenCache
+	tokenClient mgmtcontrollers.TokenClient
+}
+
+// Sync ClusterAuthToken back to Token.
+func (h *clusterAuthTokenHandler) sync(key string, clusterAuthToken *clusterv3.ClusterAuthToken) (runtime.Object, error) {
```

Comment:
So in this function, we repeatedly don't return the error, opting instead to log it but move on. For example if the `tokenCache` fails for some reason we don't re-enqueue the `clusterAuthToken` to be synced again. 

Couldn't this lead to the tokens not being updated with the `lastUsedTime` since you're deliberately avoiding the built in retry mechanism?

Reply:
Great question!

> Couldn't this lead to the tokens not being updated with the lastUsedTime since you're deliberately avoiding the built in retry mechanism?

Of course it could. This was deliberate initial decision in line with what we do on the authentication path because we don't use (at least yet) the lastUsedAt for anything important other than displaying.

Then again I'm **quite** open to making it to re-enqueue here.

Reply:
On second thought retries may not be that bad and they're out of the critical path.